### PR TITLE
Jose

### LINKED ISSUE #11: Use-Case and Misuse-case 1
For each use-case, derive security requirements using misuse case analysis. 
The misusers should be contextualized in your environment of operation and relevant to the interactions identified above. Use names for mis-users that help the reader understand their motives, resources, attack of choice, and the available access to the system to carry out their attack.
Iterate between use and misuse cases to elaborate additional security functions. Embed the final diagram in your report.
Diagraming Advice: Use proper notation as discussed in class materials. Use cases should focus on software features. Go back and forth between use case and misuse case analysis until specific functional security requirements can be identified that are implemented by the software. For misuse cases, prioritize mitigations that are implemented in the OSS project, instead of the environment. Make sure all misuse cases are addressed by use cases. Arrange the diagram to reduce clutter

### PR DESCRIPTION
**Pull request by Jbibianochavez**
First draft of my Use-Case.

### CODE REVIEW

File: Requirements for SSE.md [hunk 1]
```diff
@@ -18,19 +18,21 @@
 
 #### TODO: Insert Use Case Name
 
-TODO: Insert use-case info
+Bob, a Home Assistant software user, recently purchased a Tesla and wants to know if it is possible to control his Tesla through Home Assistant. He notices that it is possible to lock your car, control climate, track sensor data, and track location. Home Assistant relies on Tesla’s API to access this information, which requires Bob’s Tesla account username and password. This feature can be configured via the GUI or manually. Bob chooses to manually configure the add-in since he is implementing some additional security features. Bob will be using a secrets.yaml file and the keyring, as recommended by the software documentation.
```
Comment (jmpinoy) [inline Requirements for SSE.md]: Should add-in be add-on?

File: Requirements for SSE.md [hunk 2]
```diff
@@ -18,19 +18,21 @@
...
 #### TODO: Misuse case Name
 
-TODO: Insert info here
+Bob goes to a get-together, Joe, a friend of a friend, is at the party. After Bob shows a party trick of being control and view his car from his home automation system, Joe notices that Bob uses Home Assistant. Joe has some cybersecurity knowledge and has used Home Assistant before. He knows that the software interfaces with a third-party API to gain the appropriate information, and the credentials are usually saved in the configuration file or a secrets.yaml file in plain text. As the night goes on, Joe scans the network and sees Bob has an SSH port forwarded and takes note of the WAN IP. After he goes home, he brute forces the simple password with a dictionary attack and gains complete access to Home Assistant because Bob did not disable root access via SSH. Joe looks over the documentation for the Tesla add-in and sees that it reports location. Before he goes any further and risk detection, he comes up with a plan to burglarize the home with some friends while Bob is at work. Joe wants to use the location services via a home assistant or gain access to the Tesla credentials to monitor Bob’s location without his knowledge.
```
Comment (jmpinoy) [inline Requirements for SSE.md]: The second sentence, ...being control and view his car from his home automation... should it be --> ...being in control and view of his car from his home automation...

Comment (Chrs987) [inline Requirements for SSE.md]: After Bob shows a party trick of being control - Slight grammar error

**Review by Chrs987 — COMMENTED** [review]
Looks good there was a slight grammar error in 'After Bob shows a party trick of being control' for your Misuse case.

File: Requirements for SSE.md (hunk 2 above)
Comment (jmpinoy) [reply Requirements for SSE.md]: add-in be add-on?

File: Requirements for SSE.md [hunk 3]
```diff
@@ -18,19 +18,21 @@
...
 
 #### Prevention/Security Requirement
 
-TODO: Assess alignment of security requirements derived from mis-use case analysis with advertised features of the open-source software. Review OSS project documentation and codebase to support your observations. Summarize your observations.
+Home Assistant recommends not allowing root access via SSH if forwarded to the public. 
+The documentation also recommends the user make sure secrets.yaml file to separate the configurations from the username and passwords. The documentation also recommends the use of a keyring to avoid storing passwords in plain text. Lastly, it is essential to secure SSH with a key rather than a password.
```
Comment (jmpinoy) [inline Requirements for SSE.md]: First sentence could maybe be --> The documentation also recommends the user to make sure that the secrets.yaml file separates the configurations from the username and passwords.

**Review by jmpinoy — COMMENTED** [review]
Made some comments just on wording really but content looks good

**Comment by jmpinoy** [comment]
on the diagram, Burglurize should be Burglarize

**Comment by Chrs987** [comment]
also put in how root access and ssh is disabled at the beginning to help prevent it that may be helpful
also a keyring will help protect the password to the Tesla API because it will not be in plain text like it would be in the config.yaml

**Review by jmpinoy — COMMENTED** [review]
on the diagram, Burglurize should be Burglarize

**Comment by Chrs987** [comment]
Also may want to talk about a man in the middle attack where the attacker tries to sniff the traffic from HA to the API

File: Requirements for SSE.md [hunk 4]
```diff
@@ -18,19 +18,21 @@
 
 #### TODO: Insert Use Case Name
```
Comment (jmpinoy) [inline Requirements for SSE.md]: Still need to update this TODO

File: Requirements for SSE.md [hunk 5]
```diff
@@ -18,19 +18,21 @@
...
 
-TODO: Insert use-case info
+Bob, a Home Assistant software user, recently purchased a Tesla and wants to know if it is possible to control his Tesla through Home Assistant. He notices that it is possible to lock your car, control climate, track sensor data, and track location. Home Assistant relies on Tesla’s API to access this information, which requires Bob’s Tesla account username and password. This feature can be configured via the GUI or manually. Bob chooses to manually configure the add-in since he is implementing some additional security features. Bob will be using a secrets.yaml file and the keyring, as recommended by the software documentation. 
 
 #### TODO: Misuse case Name
```
Comment (jmpinoy) [inline Requirements for SSE.md]: and this TODO

File: Requirements for SSE.md (hunk 2 above)
Comment (Jbibianochavez) [reply Requirements for SSE.md]: This has been addressed, thank you for catching this.

File: Requirements for SSE.md (hunk 1 above)
Comment (Jbibianochavez) [reply Requirements for SSE.md]: Add-in has been changed to add-on.

File: Requirements for SSE.md (hunk 2 above)
Comment (Jbibianochavez) [reply Requirements for SSE.md]: This has been addressed, thank you for the revisions. The second sentece should be --> being able to control and view his car from the automation...

**Comment by Jbibianochavez** [comment]
> also put in how root access and ssh is disabled at the beginning to help prevent it that may be helpful
> also a keyring will help protect the password to the Tesla API because it will not be in plain text like it would be in the config.yaml

Good point, I read about this but the documentation did not specify this was by default. I have edited the use-case to account for this. For this situation Bob disables this feature for convenience.

**Comment by Chrs987** [comment]
Changes look good, approving to push to master.

**Review by Chrs987 — APPROVED** [review]
Changes look good approving to push to master

**Review by jmpinoy — APPROVED** [review]
Looks good

**Pull request merged**